src/features/auth/components/WithAuth.tsx

```diff
@@ -0,0 +1,17 @@
+'use client';
+
+import { useAuth } from '@processes/auth';
+import { PropsWithChildren, useEffect, useState } from 'react';
+
+function WithAuth({ children }: PropsWithChildren) {
+  const { authorized } = useAuth();
+  const [isVisible, setIsVisible] = useState(true);
+
+  useEffect(() => {
+    setIsVisible(authorized);
+  }, [authorized]);
+
+  return <>{isVisible ? children : null}</>;
```

Comment:
With the current `isVisible` name, you need to search for how state is changed to actually understand the logic here (it's pretty simple, but anyway it requires some cognitive load to process it). Maybe it makes sense to rename the state to `isAuthorized`, so there will be a single state, so you don't need to think how `isVisible` and `authorized` are connected

Reply:
yes, isAuthorized would be a more best name for the state